cubic-server/world_storage/ChunkColumn.hpp

```diff
@@ -105,6 +105,8 @@ class ChunkColumn {
     constexpr std::array<Section, NB_OF_SECTIONS> &getSections() { return _sections; }
     constexpr const std::array<Section, NB_OF_SECTIONS> &getSections() const { return _sections; }
 
+    const std::shared_ptr<Dimension> getDimension() const { return _dimension; }
```

Comment:
```suggestion
    const std::shared_ptr<const Dimension> getDimension() const { return _dimension; }
```
`const std::shared_ptr<thing>` -> the pointer is const
`std::shared_ptr<const thing>` -> the object the pointer is pointing to is const

And shared_ptr has a const copy constructor from a non-const version, so this is the only thing you need to change